### ISSUE TITLE
Hosting Hastebin on Virtual Directory using NGINX

### ISSUE THREAD

**Issue by Vexentric** [seejohnrun/haste-server#156]
I have my main website on the / location for my website and I wanted to host hastebin on /paste.
I tried it using ngix with the following location config:
`location /paste {
        proxy_pass http://localhost:8080;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_connect_timeout 150;
        proxy_send_timeout 100;
        proxy_read_timeout 100;
        proxy_buffers 4 32k;
        client_max_body_size 8m;
        client_body_buffer_size 128k;
    }`

However, it just turns out looking like this;
https://gyazo.com/0f50b9261a43a6cdc9e21d043190d7d6

Could someone help me fix this? (If it is possible to do)

**Comment by Logic-gate** [comment]
I am not a nginx guy, but try adding the `^~` modifier. 

Check out https://github.com/seejohnrun/haste-server/wiki/Installation#nginx and http://nginx.org/en/docs/http/ngx_http_core_module.html#location

**Comment by Vexentric** [comment]
It works now! Sadly the actual pasting part doesnt :(

I get this message: https://gyazo.com/0f237d18d027970f6b4bcc91b709f9d2

**Comment by Logic-gate** [comment]
Do you have redis running?

**Comment by Vexentric** [comment]
No, I'm using a file storage system. If needed I can switch to redis.

**Comment by Logic-gate** [comment]
I'd suggest moving to redis as you will have the added advantage of setting expiration. But if you want file storage, change your logging to `verbose` if you changed it and paste the log. I think it's a file permission issue, not sure without looking at the log.

**Comment by vwbusguy** [comment]
I'm using haste server behind an nginx proxy.  I didn't have to do anything elaborate:

```
server{
	listen 80;
	server_name the.listening.domain;

	location /{
		proxy_pass "http://localhost:7777/";
	}
}
```
I have a redis backend as well, but that was inconsequential as nginx doesn't need to know about it.

**Comment by Logic-gate** [comment]
@Vexentric _Ping_

**Comment by didyouexpectthat** [comment]
Hi,
I use nginx with SSL redirect (on vestacp, but you need profiles setup to do the proxy stuff).
My SSL config is :
```
server {
    listen      $IP:443;
    server_name $servername;
    ssl         on;
    ssl_certificate      /home/$user/conf/web/ssl.$servername.pem;
    ssl_certificate_key  /home/$user/conf/web/ssl.$servername.key;
    error_log  /var/log/apache2/domains/$servername.error.log error;

    location / {
        proxy_pass      http://$IP:7777;
        location ~* ^.+\.(jpeg|jpg|png|gif|bmp|ico|svg|tif|tiff|css|js|htm|html|ttf|otf|webp|woff|txt|csv|rtf|doc|docx|xls|xlsx|ppt|pptx|odf|odp|ods|odt|pdf|psd|ai|eot|eps|ps|zip|tar|tgz|gz|rar|bz2|7z|aac|m4a|mp3|mp4|ogg|wav|wma|3gp|avi|flv|m4v|mkv|mov|mpeg|mpg|wmv|exe|iso|dmg|swf)$ {
            root           /home/$user/web/$servername/public_html;
            access_log     /var/log/apache2/domains/$servername.log combined;
            access_log     /var/log/apache2/domains/$servername.bytes bytes;
            expires        max;
            try_files      $uri @fallback;
        }
    }

    location /error/ {
        alias   /home/$user/web/$servername/document_errors/;
    }

    location @fallback {
        proxy_pass      http://$IP:7777;
    }

    location ~ /\.ht    {return 404;}
    location ~ /\.svn/  {return 404;}
    location ~ /\.git/  {return 404;}
    location ~ /\.hg/   {return 404;}
    location ~ /\.bzr/  {return 404;}

}
```

**Comment by ajmar** [comment]
This still seems to be an issue. When trying to put Hastebin under a virtual directory such as /hastebin/, references in the frontend seem to be absolute assuming hastebin is at /. Setting "host" in config.js to "127.0.0.1/hastebin" also doesn't solve the issue (a solution that was required for me on another similar application). The solution (hack) proposed [here](https://github.com/seejohnrun/haste-server/wiki/Installation#nginx) is no longer an option since the repository holding that patch was deleted.

**Comment by filipechagas** [comment]
The app now supports baseUrl: https://github.com/toptal/haste-server/pull/140
